exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -5,30 +5,41 @@ package prometheusremotewriteexporter // import "github.com/open-telemetry/opent
 
 import (
 	"errors"
+	"math"
 	"sort"
 
 	"github.com/prometheus/prometheus/prompb"
 )
 
+// Track batch sizes sent to avoid over allocating huge buffers.
+// This helps in the case where large batches are sent to avoid allocating too much unused memory
+var nextTimeSeriesBufferSize = math.MaxInt
```

Comment:
there can be multiple prw exporter instances, so you should add this to the `prwExporter` struct, and make the functions that use these variables functions on `prwExporter`